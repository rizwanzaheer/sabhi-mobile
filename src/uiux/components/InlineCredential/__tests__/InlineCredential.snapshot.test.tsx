import * as React from 'react'
import { render, fireEvent, act } from 'react-native-testing-library'
import InlineCredentialInput from '../InlineCredential'

jest.mock('react-native-vector-icons/Ionicons', () => 'Icon')

describe('Component(snapshots): Inline Credential', () => {
  it('should render with required props', () => {
    const onCreate = jest.fn()
    const tree = render(<InlineCredentialInput claimType={'TEST_CLAIM'} onCreate={onCreate} />)

    expect(tree).toMatchSnapshot()
  })

  it('should fire submit event', () => {
    const onCreate = jest.fn()
    const tree = render(<InlineCredentialInput claimType={'TEST_CLAIM'} onCreate={onCreate} />)

    act(() => {
      fireEvent.changeText(tree.getByTestId('inline_credential_input'), 'TEST_CLAIM_VALUE')
    })

    fireEvent.press(tree.getByTestId('inline_credential_submit'))

    expect(onCreate).toBeCalledWith('TEST_CLAIM_VALUE')
    expect(tree).toMatchSnapshot()
  })
})
