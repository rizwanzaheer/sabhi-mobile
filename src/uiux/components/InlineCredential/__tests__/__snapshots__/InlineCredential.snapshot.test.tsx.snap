// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(snapshots): Inline Credential should fire submit event 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": undefined,
      "borderRadius": undefined,
      "flex": undefined,
      "flexDirection": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": 15,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "padding": undefined,
      "paddingBottom": 15,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": 15,
      "width": undefined,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#BBBBBB",
        "flexWrap": "wrap",
        "fontSize": 14,
      }
    }
  >
    Self sign a 
    <Text
      style={
        Object {
          "flexWrap": "wrap",
          "fontWeight": "bold",
        }
      }
    >
      TEST_CLAIM
    </Text>
     credential
  </Text>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "borderColor": "#CCCCCC",
        "borderRadius": 5,
        "borderWidth": 1,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": 15,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": 15,
        "padding": 15,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#BBBBBB",
            "flexWrap": "wrap",
            "fontSize": 14,
          }
        }
      >
        TEST_CLAIM
      </Text>
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "center",
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": 5,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="sentences"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="Enter TEST_CLAIM"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#000000",
            "flex": 1,
            "fontSize": 19,
            "padding": 0,
          }
        }
        testID="inline_credential_input"
        underlineColorAndroid="transparent"
        value="TEST_CLAIM_VALUE"
      />
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": undefined,
            "backgroundColor": undefined,
            "borderRadius": undefined,
            "flex": undefined,
            "flexDirection": undefined,
            "height": undefined,
            "justifyContent": undefined,
            "margin": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "width": undefined,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="inline_credential_submit"
        >
          <Icon
            color="#0078ff"
            name="ios-arrow-dropright"
            size={28}
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Component(snapshots): Inline Credential should render with required props 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": undefined,
      "borderRadius": undefined,
      "flex": undefined,
      "flexDirection": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": 15,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "padding": undefined,
      "paddingBottom": 15,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": 15,
      "width": undefined,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#BBBBBB",
        "flexWrap": "wrap",
        "fontSize": 14,
      }
    }
  >
    Self sign a 
    <Text
      style={
        Object {
          "flexWrap": "wrap",
          "fontWeight": "bold",
        }
      }
    >
      TEST_CLAIM
    </Text>
     credential
  </Text>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "borderColor": "#CCCCCC",
        "borderRadius": 5,
        "borderWidth": 1,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": 15,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": 15,
        "padding": 15,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#BBBBBB",
            "flexWrap": "wrap",
            "fontSize": 14,
          }
        }
      >
        TEST_CLAIM
      </Text>
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": "row",
          "height": undefined,
          "justifyContent": "center",
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": 5,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="sentences"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="Enter TEST_CLAIM"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#000000",
            "flex": 1,
            "fontSize": 19,
            "padding": 0,
          }
        }
        testID="inline_credential_input"
        underlineColorAndroid="transparent"
      />
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": undefined,
            "backgroundColor": undefined,
            "borderRadius": undefined,
            "flex": undefined,
            "flexDirection": undefined,
            "height": undefined,
            "justifyContent": undefined,
            "margin": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "width": undefined,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="inline_credential_submit"
        >
          <Icon
            color="#BBBBBB"
            name="ios-arrow-dropright"
            size={28}
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
